### ISSUE TITLE
Difference in total time of display=True and display=False

### ISSUE THREAD

**Issue by nihal-rao** [awwong1/torchprof#7]
Thanks for this wonderful tool!
However for the same model, I see some difference in the times when using `prof.display(show_events=False)` and `prof.display(show_events=True)`.
When using `prof.display(show_events=False)`, this is a part of the output :
```
├── proposal_generator  |                |           |           
│├── anchor_generator   |                |           |           
││└── cell_anchors      |        0.000us |   0.000us |    0.000us
│├── rpn_head           |                |           |           
││├── conv              |      569.006us |   2.113ms |  167.045ms
││├── objectness_logits |      389.972us |   1.400ms |    6.801ms
││└── anchor_deltas     |      342.629us |   1.210ms |    7.177ms

```
**The above shows that the conv layer under rpn_head takes 167.045 ms.**
However when I use `prof.display(show_events=True)`, the same part of the output now becomes:
```
│├── rpn_head                    |                |           |           
││├── conv                       |                |           |           
│││├── conv2d                    |        4.283us | 102.069us |  433.152us
│││├── convolution               |        3.701us |  97.786us |  430.080us
│││├── _convolution              |        9.287us |  94.085us |  424.960us
│││├── contiguous                |        1.963us |   1.963us |    2.048us
│││└── cudnn_convolution         |       82.835us |  82.835us |  414.720us
││├── objectness_logits          |                |           |           
│││├── conv2d                    |        3.989us |  74.004us |   87.040us
│││├── convolution               |        3.754us |  70.015us |   82.944us
│││├── _convolution              |       13.054us |  66.261us |   78.848us
│││├── contiguous                |        2.232us |   2.232us |    0.512us
│││└── cudnn_convolution         |       50.975us |  50.975us |   65.536us
││├── anchor_deltas              |                |           |           
│││├── conv2d                    |        4.255us |  69.547us |   80.896us
│││├── convolution               |        3.742us |  65.292us |   75.776us
│││├── _convolution              |        9.442us |  61.550us |   71.680us
│││├── contiguous                |        2.119us |   2.119us |    2.048us
│││└── cudnn_convolution         |       49.989us |  49.989us |   60.416us

```
**Here however by summing the values under conv_layer of rpn_head, the time taken is 1.7us.**
Why is there such a huge difference between these two values? Shouldn't they be similar?

**Comment by awwong1** [comment]
The problem is that in the first block, the CUDA total time is 167.045 ms, while in the second block, the CUDA total time is 1.7ms, correct?

What happens when:
- you run these multiple times?
- `prof.display(show_events=True)` is run before `prof.display(show_events=False)`

I do know that the CUDA memory allocator and profiler requires a warmup, so I typically discard the first profile result and perform an average over ~100 runs, accounting for temperature throttling.

**Comment by nihal-rao** [comment]
- It doesn't change , even when it is run multiple times and when `prof.display(show_events=True)` is run before `prof.display(show_events=False)`.
- Also the only difference between the results of `prof.display(show_events=True)` and `prof.display(show_events=False)` is the rpn head as described above. For all other layers the times match closely.

Surprisingly , the entire model itself takes on average ~140 ms to run, whereas when `prof.display(show_events=False)` the time taken for the conv operation itself under rpn_head (as shown above) is 167.045 ms.

For reference, I am profiling a faster rcnn model provided in detectron2's model zoo.

**Comment by awwong1** [comment]
I was able to [replicate this issue and I can confirm it is an existing limitation of torchprof](https://gist.github.com/awwong1/4e7bfbddbadaa85884fedab049118443).

```python3
paths = [("GeneralizedRCNN", "proposal_generator", "rpn_head", "conv"),]

with torchprof.Profile(predictor.model, paths=paths, use_cuda=True) as prof:
    predictor(im)
print(prof.display(show_events=False))

print("=" * 40)
print(prof.display(show_events=True))

trace, event_lists_dict = prof.raw()
# trace[262] # Trace(path=('GeneralizedRCNN', 'proposal_generator', 'rpn_head', 'conv'), leaf=True, module=Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1)))

for evl_run in event_lists_dict[trace[262].path]:
    print(evl_run)
```

In this model, `GeneralizedRCNN/proposal_generator/rpn_head/conv'` is run 5 times on a single forward pass. The show_events=True code incorrectly only looks at one of these 5 runs when outputting the totals.

Thank you for opening this issue, @nihal-rao! I will create a fix shortly.

**Comment by awwong1** [comment]
Addressed as of https://github.com/awwong1/torchprof/releases/tag/v1.1.0

Please feel free to comment again if this issue is not resolved for you.
